Sofa/Component/IO/Mesh/tests/MeshSTLLoader_test.cpp

```diff
@@ -0,0 +1,93 @@
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+
+#include <sofa/testing/BaseTest.h>
+#include <sofa/helper/system/FileRepository.h>
+
+#include <SofaGeneralLoader/MeshSTLLoader.h>
+#include <sofa/helper/BackTrace.h>
+
+using namespace sofa::component::loader;
+using sofa::testing::BaseTest;
+using sofa::helper::BackTrace;
+
+namespace sofa
+{
+namespace meshstlloader_test
+{
+
+    int initTestEnvironment()
+    {
+        BackTrace::autodump();
+        return 0;
+    }
+    int s_autodump = initTestEnvironment();
+
+
+    class MeshSTLLoaderTest : public BaseTest,
+                              public MeshSTLLoader
+    {
+    public:
+
+        MeshSTLLoaderTest()
+        {}
+
+        /**
+         * Helper function to check mesh loading.
+         * Compare basic values from a mesh with given results.
+         */
+        void loadTest(std::string filename, int nbPositions, int nbEdges, int nbTriangles, int nbQuads, int nbPolygons,
+                      int nbTetra, int nbHexa, int nbNormals)
+        {
+            this->setFilename(sofa::helper::system::DataRepository.getFile(filename));
+
+            EXPECT_TRUE(this->load());
+            EXPECT_EQ((size_t)nbPositions, this->d_positions.getValue().size());
+            EXPECT_EQ((size_t)nbEdges, this->d_edges.getValue().size());
+            EXPECT_EQ((size_t)nbTriangles, this->d_triangles.getValue().size());
+            EXPECT_EQ((size_t)nbQuads, this->d_quads.getValue().size());
+            EXPECT_EQ((size_t)nbPolygons, this->d_polygons.getValue().size());
+            EXPECT_EQ((size_t)nbTetra, this->d_tetrahedra.getValue().size());
+            EXPECT_EQ((size_t)nbHexa, this->d_hexahedra.getValue().size());
+            EXPECT_EQ((size_t)nbNormals, this->d_normals.getValue().size());
+        }
+
+    };
+
+    /** MeshGmshLoader::load()
+    * For a given meshes check that imported data are correct
+    */
+    TEST_F(MeshSTLLoaderTest, LoadCall)
+    {
+        //Check loader high level result for one file
+        this->setFilename(sofa::helper::system::DataRepository.getFile("mesh/circle_knot_ascii.stl"));
+        EXPECT_TRUE(this->load());
+
+        // Testing number of : nodes/positions, edges, triangles, quads, polygons, tetra, hexa, normals
+        // for several STL meshes in share/mesh/
+        loadTest("mesh/circle_knot_ascii.stl", 6144, 0, 12288, 0, 0, 0, 0, 12288);
+        loadTest("mesh/dragon.stl", 1190, 0, 2526, 0, 0, 0, 0, 2526);
+        loadTest("mesh/pliers_binary.stl", 5356, 0, 10712, 0, 0, 0, 0, 10712); //NB : Gmsh detects 10708 faces. Is that normal ?
```

Comment:
I highlight the difference you found because it may indicate a problem. This could be added to an issue.
```suggestion
        loadTest("mesh/pliers_binary.stl", 5356, 0, 10712, 0, 0, 0, 0, 10712); //NB : Gmsh detects 10708 faces
```

Is that normal ? => No

Reply:
Same for blender, it indicates 10708 triangles, 5356 vertices and no quads.
Looking at the scene in runSofa, the 4 additional triangles are not obvious... I have no idea how to debug and find the 4 wrong triangles inside the 10712....
I'm not even sure an issue is worth it.